paimon-core/src/main/java/org/apache/paimon/table/source/snapshot/IncrementalStartingScanner.java

```diff
@@ -111,4 +125,13 @@ private List<DataSplit> readDeltaSplits(SnapshotReader reader, Snapshot s) {
         }
         return (List) reader.withSnapshot(s).withMode(ScanMode.DELTA).read().splits();
     }
+
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    private List<DataSplit> readChangeLogSplits(SnapshotReader reader, Snapshot s) {
+        if (s.commitKind() != CommitKind.COMPACT) {
```

Comment:
Maybe we should not ignore APPEND for changelog.

Reply:
Yes you are right, APPEND is also need